src/ClientHandle.cpp

```diff
@@ -1243,13 +1246,32 @@ void cClientHandle::HandleBlockDigFinished(int a_BlockX, int a_BlockY, int a_Blo
 		}
 	}
 
+	if (!m_Player->IsGameModeCreative() && !cBlockInfo::IsOneHitDig(a_OldBlock))
+	{
+		// Fix for very fast tools.
+		m_BreakProgress += m_Player->GetPlayerRelativeBlockHardness(a_OldBlock);
+		if (m_BreakProgress < FASTBREAK_PERCENTAGE)
+		{
+			LOGD("Break progress of player %s was less than expected: %f < %f\n", m_Player->GetName().c_str(), m_BreakProgress * 100, FASTBREAK_PERCENTAGE * 100);
+			// AntiFastBreak doesn't agree with the breaking. Bail out. Send the block back to the client, so that it knows:
+			m_Player->GetWorld()->SendBlockTo(a_BlockX, a_BlockY - 1, a_BlockZ, m_Player);  // Bug with doors and things like that.
+			m_Player->GetWorld()->SendBlockTo(a_BlockX, a_BlockY, a_BlockZ, m_Player);
+			m_Player->GetWorld()->SendBlockTo(a_BlockX, a_BlockY + 1, a_BlockZ, m_Player);  // Bug with doors and things like that.
```

Comment:
Never mind.  I took a closer look and `onBlockHarvested` _is_ called clientside as well as serverside.  So you do need to resend those for:

* Pistons, when the head is broken (the base is also broken)
* Double plants, maybe... maybe not
* Doors (top/bottom)
* Beds (both halves)

The client also deletes NBT for flower pots and skulls.  So if the block is resent, you need to resend that too.

I'd recommend explicitly handling those blocks rather than doing a general case.